src/metabase/models/query/permissions.clj

```diff
@@ -168,7 +168,7 @@
             (format "Expected QP preprocessing to return legacy MBQL query, got: %s" (pr-str legacy-query)))
     (legacy-mbql-required-perms legacy-query perms-opts)))
 
-(defn required-perms
+(defn required-perms-for-query
```

Comment:
Renamed this so it doesn't conflict with the `required-perms` arg of `check-data-perms` which was causing some issues when testing this in the REPL